plugins/tor/tor_

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/python2
+
+from __future__ import print_function
+import collections
+import json
+import os
+import stem
+import stem.control
+import sys
+
+default_torgeoippath = "/usr/share/GeoIP/GeoIP.dat"
+default_torcachefile = 'munin_tor_country_stats.json'
+default_torport = 9051
+default_torsocket = '/var/run/tor/control'
+default_torconnectmethod = 'port'
+
+#%# family=auto
+#%# capabilities=autoconf suggest
+
+def simplify(cn):
+    """Simplify country name"""
+    cn = cn.replace(' ', '_')
+    cn = cn.replace("'", '_')
+    cn = cn.split(',', 1)[0]
+    return cn
+
+
+class ConnectionError(Exception):
+    """Error connecting to the controller"""
+
+
+class AuthError(Exception):
+    """Error authenticating to the controller"""
+
+
+def authenticate(controller):
+    try:
+        controller.authenticate()
+        return
+    except stem.connection.MissingPassword:
+        pass
+
+    try:
+        password = os.environ['torpassword']
+    except KeyError:
+        raise AuthError("Please configure the 'torpassword' "
+                        "environment variable")
+
+    try:
+        controller.authenticate(password=password)
+    except stem.connection.PasswordAuthFailed:
+        print("Authentication failed (incorrect password)")
```

Comment:
Should this be redirected to `sys.stderr`?

btw: wouldn't an `AuthError` be suitable here, as well?